trie/baseNode.go

```diff
@@ -0,0 +1,47 @@
+package trie
+
+import (
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"sync"
+)
+
+type baseNode struct {
+	mutex  sync.RWMutex
+	hash   []byte
+	dirty  bool
+	marsh  marshal.Marshalizer
+	hasher hashing.Hasher
+}
+
+func (bn *baseNode) getHash() []byte {
+	return bn.hash
+}
+
+func (bn *baseNode) setGivenHash(hash []byte) {
+	bn.hash = hash
+}
+
+func (bn *baseNode) isDirty() bool {
+	return bn.dirty
```

Comment:
i think rootManager is accessing this so it needs protection here

Reply:
Indeed, will refactor mutex usage in a future PR